todo_app/data/services/github_service.py

```diff
@@ -1,40 +1,55 @@
+import logging
+
 import requests
 from flask import current_app, redirect
 from werkzeug.exceptions import abort
 
 
 def get_access_token(auth_code):
-    headers = {
-        'Accept': 'application/json'
-    }
+    try:
+        headers = {
+            'Accept': 'application/json'
+        }
+
+        data = {
+            'client_id': current_app.config.get("CLIENT_ID"),
+            'client_secret': current_app.config.get("CLIENT_SECRET"),
+            'code': auth_code
+        }
+        response = requests.post('https://github.com/login/oauth/access_token', data=data, headers=headers)
+
+        json_data = response.json()
 
-    data = {
-        'client_id': current_app.config.get("CLIENT_ID"),
-        'client_secret': current_app.config.get("CLIENT_SECRET"),
-        'code': auth_code
-    }
-    response = requests.post('https://github.com/login/oauth/access_token', data=data, headers=headers)
+        if 'error' in json_data and json_data['error'] == "bad_verification_code":
```

Comment:
An alternative way of writing the same - the `.get` method for dictionaries returns None rather than throwing an error if nothing matches (or returns a default if we offer one as a second argument):
```suggestion
        if json_data.get('error') == "bad_verification_code":
```